src/DIPS.Xamarin.UI/Controls/Toast/Toast.cs

```diff
@@ -0,0 +1,274 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using DIPS.Xamarin.UI.Internal.Xaml;
+using Xamarin.Essentials;
+using Xamarin.Forms;
+
+namespace DIPS.Xamarin.UI.Controls.Toast
+{
+    /// <summary>
+    ///     Toast control that would appear on top of the presented view
+    /// </summary>
+    public class Toast
+    {
+        private Toast()
+        {
+            Application.Current.PageAppearing -= OnPageAppearing;
+            Application.Current.PageAppearing += OnPageAppearing;
+
+            Application.Current.PageDisappearing -= OnPageDisappearing;
+            Application.Current.PageDisappearing += OnPageDisappearing;
+        }
+
+        /// <summary>
+        ///     Get the current instance of the Toast control
+        /// </summary>
+        public static Toast Current { get; } = new Toast();
+
+        private CancellationTokenSource CancellationSource { get; set; } = new CancellationTokenSource();
+        private Dictionary<string, Grid> ToastContainers { get; } = new Dictionary<string, Grid>();
+        private ToastLayout ToastLayout { get; set; }
```

Comment:
I think it would be better to add an internal Toast class which has properties, so that race conditions don't override on different toasts.

Reply:
The `ToastLayout` and `ToastOptions` is moved to `ToastView` class.

Do you feel we got to move something additional too?